sympy/functions/special/delta_functions.py

```diff
@@ -67,6 +67,8 @@ def eval(cls, arg, k=0):
             return S.NaN
         if arg.is_positive or arg.is_negative:
             return S.Zero
+        if fuzzy_not(im(arg).is_zero):
+            raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
```

Comment:
This is really a good way of inputting multiline messages. Thanks.
I am adding this in a new PR.